### PR TITLE
Convert rib-test to Kotlin

### PR DESCRIPTION
**Pull request by jbarr21**
Migrates the `rib-test` module to Kotlin as part of #412

### CODE REVIEW

File: android/libraries/rib-test/src/main/kotlin/com/uber/rib/core/RouterHelper.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+   * @param router the [Router].
+   * @param mode The mockito verification mode. ie. `times(1)`.
+   */
+  @JvmStatic
+  open fun verifyDetached(router: Router<*>, mode: VerificationMode) {
+    verify(router, mode).dispatchDetach()
+  }
+
+  /**
+   * Verifies that the [Router] is detached.
```
Comment (ekager) [inline android/libraries/rib-test/src/main/kotlin/com/uber/rib/core/RouterHelper.kt:150]: nit - Should this be verify is _not_ detached?

File: android/libraries/rib-test/src/main/kotlin/com/uber/rib/core/RouterHelper.kt [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+   * @param order [InOrder] for ordered verification.
+   * @param router the [Router].
+   */
+  @JvmStatic
+  open fun verifyAttached(order: InOrder, router: Router<*>) {
+    order.verify(router).dispatchAttach(or(isNull(Bundle::class.java), isA(Bundle::class.java)), anyString())
+  }
+
+  /**
+   * Verified that the [Router] is attached with a sepcific tag.
```
Comment (ekager) [inline android/libraries/rib-test/src/main/kotlin/com/uber/rib/core/RouterHelper.kt:85]: nit - specific*

**Review by ekager — APPROVED** [review]
🎇

File: android/libraries/rib-test/src/main/kotlin/com/uber/rib/core/RouterHelper.kt (hunk 1 above)
Comment (jbarr21) [reply android/libraries/rib-test/src/main/kotlin/com/uber/rib/core/RouterHelper.kt]: c2057701daf5c855b12768f8765c39b0ef2f0b13

File: android/libraries/rib-test/src/main/kotlin/com/uber/rib/core/RouterHelper.kt (hunk 2 above)
Comment (jbarr21) [reply android/libraries/rib-test/src/main/kotlin/com/uber/rib/core/RouterHelper.kt]: c2057701daf5c855b12768f8765c39b0ef2f0b13

**Pull request merged**